firestore.rules

```diff
@@ -13,8 +13,17 @@ service cloud.firestore {
       allow read: if request.auth.uid != null && request.auth.token.firebase.sign_in_provider != 'anonymous';
     }
     match /sources/{source}/answers/{document=**} {
```

Comment:
Having just been in this space for the CLUE teacher support work, let me double-check my understanding here. The `{document=**}` means the rules apply to any document below `answers` at any hierarchical depth. Several of the rules apply to the content of the document, e.g. `string(resource.data.platform_user_id)`. Presumably this means that every document below `answers` has those fields, which I'm guessing is not a particularly deep or granular hierarchy.

Reply:
> which I'm guessing is not a particularly deep or granular hierarchy.

Yes, we just have flat answer JSON documents, there are no subcollections at the moment. And yes, we make assumptions about some fields being available, but it seems fine to me.